client/js/_author/selectors/bank_navigator.js

```diff
@@ -0,0 +1,49 @@
+import _ from 'lodash';
+import { createSelector }  from 'reselect';
+import * as common from './common';
+import { _assessment } from './assessment';
+
+export const path = state => state.bankNavigation.location;
+
+
+export const currentBankId = createSelector(
+  path,
+  (_path) => {
+    if (!_.isEmpty(_path)) return _.last(_path).id;
+    return null;
+  }
+);
+
+export const rawAssessments = createSelector(
+  common.assessments,
+  currentBankId,
+  (_rawAssessments, _currentBankId) => _.get(_rawAssessments, _currentBankId, {})
+);
+
+export const bankAssessments = createSelector(
+rawAssessments,
+common.settings,
+(_rawAssessments, _settings) => {
+  const transformedAssessments =
+    _.map(_rawAssessments, assessment => _assessment(assessment, [], _settings));
+  const assessmentHash = _.reduce(
+    transformedAssessments,
+    (assessments, assessment) => {
+      assessments[assessment.id] = assessment;
+      return assessments;
+    }, {});
+  return assessmentHash;
+});
+
+
+export const banks = createSelector(
+  path,
+  state => state.banks,
+  (_path, _banks) => {
```

Comment:
Yeah I agree. I took a long time yesterday trying to figure out what it was doing ha.